### DISCUSSION TITLE
Labs Feedback: @lit-labs/ssr

### DISCUSSION THREAD

**Discussion by augustjk** [lit/lit#3353] [Labs Feedback]
The `@lit-labs/ssr` package is a package for server-side rendering Lit templates and components.

- Status: **Under development**

- Major known issues:
  - https://github.com/lit/lit/issues/3365
  - https://github.com/lit/lit/issues/3219
  - https://github.com/lit/lit/issues/2469
  - https://github.com/lit/lit/issues/3080
  - https://github.com/lit/lit/issues/3122
  - https://github.com/lit/lit/issues/1434
- Issues: [[labs/ssr]](https://github.com/lit/lit/issues?q=is%3Aissue+is%3Aopen+%5Blabs%2Fssr%5D+in%3Atitle)
- Docs: [Server Rendering on lit.dev](https://lit.dev/docs/ssr/overview/)
- npm: [@lit-labs/ssr](https://www.npmjs.com/package/@lit-labs/ssr)

Please try `@lit-labs/ssr` and tell us about your experience. Leave comments with your thoughts; highlight what issues or missing features you're blocked on; give a link to a project that uses it; etc.

And so we can have some numbers to look at, you can add reactions to this comment to indicate your experience with this package:

🚀 I have tried this package and it works great. I need it. Ship it!
👍 I have tried this package and it seems like a good idea.
😕 I have tried this package and would use it, but it has serious bugs or missing features.
👀 I am interested, but haven't tried the package yet.
👎 I'm not interested in this package.

Note: other reactions like 😄, 🎉, and ❤️ don't mean anything as part of this scheme. You probably mean, and please use, 🚀, 👍, or 👀.

**Comment by vekunz** [comment]
I recently discovered this plugin and it is exactly what we need at work! For several projects, we use Vue/Nuxt or Svelte/SvelteKit. Since there is no integration into Svelte/SvelteKit, I created a proof of concept for SvelteKit in this repository https://github.com/vekunz/vite-lit-ssr-poc. I am not sure if the hydration of Lit works correctly, but at least something seems to happen 😄 .

Besides this, it was very easy to integrate the Lit renderer into the build pipeline.

**Reply by vekunz** [reply]
After trying out `nuxt-ssr-lit` (which is just works) I noticed that with Svelte 5, the hydration of Lit SSR on the client side just does not work. In Svelte 4 there is "just" a major issue with hydration. But I am not that into Svelte and Lit to further analyze this (but I updated my poc repo, maybe someone else can do this).
